SpellChecker.java

```diff
@@ -8,27 +8,103 @@ public static void main(String[] args) {
 		String[] dictionary = readDictionary("dictionary.txt");
 		String correction = spellChecker(word, threshold, dictionary);
 		System.out.println(correction);
+		System.out.println(tail (word));
+		System.out.println (levenshtein("SPELL","spell"));
 	}
 
-	public static String tail(String str) {
+	public static String tail(String str)
+	{
+		String newStr = str;
+
+		if (newStr.length() ==1)
+		{	
+			return null; 
```

Comment:
You were required to return empty string, not null - these are different values.
```suggestion
			return ""; 
```